internal/catalog/internal/controllers/workloadhealth/controller.go

```diff
@@ -25,95 +27,49 @@ var (
 	errNodeHealthConditionNotFound = fmt.Errorf("Node health status is missing the %s condition", nodehealth.StatusConditionHealthy)
 )
 
-// The NodeMapper interface is used to provide an implementation around being able to
-// map a watch event for a Node resource and translate it to reconciliation requests
-// for all Workloads assigned to that node.
-type NodeMapper interface {
-	// MapNodeToWorkloads will take a Node resource and return controller requests
-	// for all Workloads associated with the Node.
-	MapNodeToWorkloads(ctx context.Context, rt controller.Runtime, res *pbresource.Resource) ([]controller.Request, error)
-
-	// TrackWorkload instructs the NodeMapper to associate the given workload
-	// ID with the given node ID.
-	TrackWorkload(workloadID *pbresource.ID, nodeID *pbresource.ID)
-
-	// UntrackWorkload instructs the Nodemapper to forget about any
-	// association it was tracking for this workload.
-	UntrackWorkload(workloadID *pbresource.ID)
-
-	// NodeIDFromWorkload is used to generate the resource ID for the Node referenced
-	// within the NodeName field of the Workload.
-	NodeIDFromWorkload(workload *pbresource.Resource, workloadData *pbcatalog.Workload) *pbresource.ID
-}
-
-func WorkloadHealthController(nodeMap NodeMapper) *controller.Controller {
-	if nodeMap == nil {
-		panic("No NodeMapper was provided to the WorkloadHealthController constructor")
-	}
+func WorkloadHealthController() *controller.Controller {
+	nodeIndexer := indexers.DecodedSingleIndexer[*pbcatalog.Workload](
+		"node",
+		index.ReferenceOrIDFromArgs,
+		nodeIndexer,
```

Comment:
It seems strange to use `nodeIndexer` on both the LHS and RHS of this given that it is declared in this line.

Reply:
Fixed it to not use intermediate variable to hold onto the indexer but inline in the Watch definition.